sklearn/preprocessing/_polynomial.py

```diff
@@ -207,9 +212,26 @@ def get_feature_names(self, input_features=None):
         -------
         output_feature_names : list of str of shape (n_output_features,)
         """
+        return self.get_feature_names_out(input_features)
+
+    def get_feature_names_out(self, input_features=None):
+        """Get output feature names for transformation.
+
+        Parameters
+        ----------
+        input_features : array-like of str or None, default=None
```

Comment:
If we return a `list` of `str` we should probably say so explicitly. But personally I think I would be in favor of returning an array of `str` objects: https://github.com/scikit-learn/scikit-learn/pull/18444/files#r697723473